src/components/Notifications/Notifications.jsx

```diff
@@ -0,0 +1,172 @@
+    }, 5000); // 5 second delay
+    timeoutId = timeId;
+  };
+
+  const removeNotificationEntry = key => {
+    setNotificationList(notificationBlocks =>
+      notificationBlocks.filter(block => block.getKey() !== key),
+    );
+  };
+
+  useEffect(() => {
+    const fetchNotificationData = async () => {
+      const [pendingAccounts, addedEvents, confirmedEvents] = await Promise.all([
+        NPOBackend.get(`/users/pending-accounts`).then(res => res.data),
+        NPOBackend.get('/published-schedule/recently-added').then(res => res.data),
+        NPOBackend.get('/published-schedule/recently-confirmed').then(res => res.data),
+      ]);
+
+      // Map MM-DD-YY string to AccountNotificationBlock
+      const accountsMap = new Map();
+
+      pendingAccounts.forEach(({ approvedOn, email, id, firstName }) => {
+        const { dateObject, formattedDateString } = getDateFromISOString(approvedOn);
+        let notificationBlock;
+
+        if (!accountsMap.has(formattedDateString)) {
+          // Create notification date block if first found entry of that day
+          const key = `accounts-${formattedDateString}`;
+          notificationBlock = new AccountNotificationBlock(dateObject, 'account', key);
+        } else {
+          // Set notification block to be existing entry
+          notificationBlock = accountsMap.get(formattedDateString);
+        }
+
+        notificationBlock.addPendingAccount(
+          id,
+          firstName,
+          email,
+          async () => {
+            await approveAccount(id);
+          },
+          async () => {
+            await declineAccount(id);
+          },
+          undoChanges,
+        );
+        accountsMap.set(formattedDateString, notificationBlock);
+      });
+
+      // Map MM-DD-YY to EventNotificationBlock
+      const eventsArray = [];
+
+      addedEvents.forEach(({ createdOn, title, id }) => {
+        const { dateObject } = getDateFromISOString(createdOn);
+        const key = `event-added-${id}`;
+        eventsArray.push(new EventNotificationBlock(dateObject, 'event', key, title, 'added'));
+      });
+
+      confirmedEvents.forEach(({ confirmedOn, title, id }) => {
+        const { dateObject } = getDateFromISOString(confirmedOn);
+        const key = `event-confirmed-${id}`;
+        eventsArray.push(new EventNotificationBlock(dateObject, 'event', key, title, 'confirmed'));
+      });
+
+      const notificationBlocks = [...Array.from(accountsMap.values()), ...eventsArray];
+
+      // Sort notification blocks in reverse chronological order,
+      // treating every block as NotificationBlock (base class) object
+      notificationBlocks.sort((notificationBlock1, notificationBlock2) => {
+        const date1 = notificationBlock1.getDate();
+        const date2 = notificationBlock2.getDate();
+
+        return date2 - date1;
+      });
+
+      setNotificationList(notificationBlocks);
+    };
+    fetchNotificationData().catch(console.error);
+  }, []);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has missing dependencies: 'approveAccount', 'declineAccount', and 'undoChanges'. Either include them or remove the dependency array.